tests/test_partial_measurements.py

```diff
@@ -0,0 +1,56 @@
+"""Tests pending experiments mechanism."""
```

Comment:
I think you mean measurements here, not experiments?
```suggestion
"""Tests for pending measurements."""
```